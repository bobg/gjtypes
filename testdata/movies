{"title": "The Godfather",
 "director": "Francis Ford Coppola",
 "year": 1972,
 "stars": ["Marlon Brando",
           "Al Pacino",
           "Robert Duvall",
           "James Caan",
           "Talia Shire",
           "John Cazale"]}

{"title": "Casablanca",
 "director": "Michael Curtiz",
 "year": 1942,
 "stars": ["Humphrey Bogart",
           "Ingrid Bergman",
           "Paul Henreid",
           "Claude Rains",
           "Peter Lorre",
           "Sydney Greenstreet",
           "Dooley Wilson",
           "Conrad Veidt"]}

{"title": "Scott Pilgrim vs. The World",
 "director": "Edgar Wright",
 "year": 2010,
 "stars": ["Michael Cera",
           "Mary Elizabeth Winstead",
           "Ellen Wong",
           "Chris Evans",
           "Aubrey Plaza",
           "Anna Kendrick",
           "Brie Larson"],
 "songs": [{"title": "Teenage Dream",
            "artist": "T. Rex"},
           {"title": "Black Sheep",
            "artist": "Metric"}]}
