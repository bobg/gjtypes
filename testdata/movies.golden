var data []*S001 // Unmarshal into this type.

type S001 struct {
	Director string   `json:"director,omitempty"`
	Songs    []*S002  `json:"songs,omitempty"`
	Stars    []string `json:"stars,omitempty"`
	Title    string   `json:"title,omitempty"`
	Year     int64    `json:"year,omitempty"`
}

type S002 struct {
	Artist string `json:"artist,omitempty"`
	Title  string `json:"title,omitempty"`
}
